docs/design-documents/tools/cmake.md

```diff
@@ -1,89 +1,85 @@
+1. Profile configuration (profile.cmake) - configures the profile from cmake/profiles directory based on the selected profile (`MBED_PROFILE`)
+1. Core configuration (core.cmake) - configures the core from cmake/cores directory based on the selected target (`MBED_CPU_CORE`)
+1. Utilities (util.cmake) - functions or macros we use
+1. Application cmake (app.cmake) - an application includes it to
+1. Each component has own CMake file (describing the component - what files are to include, target/feature/component selection based on target, etc)
 
 The next sections will describe static CMake files within Mbed OS Core repository.
 
 ### 1. Boilerplate CMake
 
-The main CMake file in Mbes OS repository provides just boilerplate for Mbed OS to be built. It described the Mbed OS tree, provides all the options we have in Mbed OS. This will be autogenerated by the tools.
+The main CMake file in Mbes OS repository provides just boilerplate for Mbed OS to be built. It described the Mbed OS tree, provides all the options we have in Mbed OS.
+
+This is not intended to be included by an application.
 
 ### 2. Toolchain CMake
 
-There are 3 toolchains, we provide the template that tools could fill in. The information is already in the build tools, we just need to extract that info and make it CMake compatible.
+All the toolchain settings are defined in cmake/toolchains cmake files.
 
-This toolchain CMake is included by the Mbed OS CMake.
+### 3. Profile CMake
 
-### 3. Module CMake
+The application profiles are defined in cmake/profiles.
 
-This file statically defines the structure of the module within Mbed OS. It's conditionally config based. We use regular CMake expressions plus extend it with own macros to conditionally include/exclude directories. The thumb of the rule, do not expose headers that are internal. We would like to avoid having everything in the include paths as we do now.
+### 4. Core CMake
 
+The core definitions are defined in cmake/cores.
 
-`add_subdirectory` always adds the directory to the main CMake.
+### 5. Utilities
 
-```
-add_subdirectory(drivers)
-```
+Functions or macros that we use within Mbed OS and might be useful for other as well.
+
+### 6. Application CMake
 
-Conditionally directories addition based on the config:
+This is the cmake that must be included by an application, using:
 
 ```
-add_subdirectory_if_config(CONFIG_REQUIRES_RTOS rtos)
+include(${MBED_ROOT}/cmake/app.cmake)
 ```
 
-Conditionally directories addition based on the target/toolchain/feature.
+### 7. Component CMake
 
+This file statically defines the structure of the component within Mbed OS. It is conditionally config based. We use regular CMake expressions plus extend it with own macros to conditionally include/exclude directories. The thumb of the rule, do not expose headers that are internal. We would like to avoid having everything in the include paths as we do now.
```

Comment:
```suggestion
This file statically defines the structure of the component within Mbed OS. It contains conditional statements that are based on the config. We use regular CMake expressions, plus extend it with our own macros, to conditionally include/exclude directories. The rule of thumb: do not expose headers that are internal. We would like to avoid having everything in the include paths as we do now.
```

Reply:
Thanks, will review all and update.